crates/cast/bin/main.rs

```diff
@@ -723,7 +725,22 @@ async fn main_args(args: CastArgs) -> Result<()> {
     fn print_tokens(tokens: &[DynSolValue]) {
         if shell::is_json() {
             let tokens: Vec<String> = format_tokens_raw(tokens).collect();
-            let _ = sh_println!("{}", serde_json::to_string_pretty(&tokens).unwrap());
+            let tokens_str: String = serde_json::to_string(&tokens).unwrap();
+            let input = tokens_str.replace("(", "[").replace(")", "]");
+            let data: Vec<String> = serde_json::from_str(&input).unwrap();
+            let re = Regex::new(r"0x[0-9a-fA-F]+").unwrap();
+            let mut result: Vec<Value> = Vec::new();
+
+            for i in data {
+                let i =
+                    re.replace_all(&i, |caps: &regex::Captures<'_>| format!(r#""{}""#, &caps[0]));
```

Comment:
This is not the right fix, instead we should have a separate function in `crates/common/fmt/src/dynamic.rs` that prints as JSON recursively instead of just the top layer.

I understand this is a more complex change, if you don't have capacity / will to do this we should open an issue for it and close this PR.